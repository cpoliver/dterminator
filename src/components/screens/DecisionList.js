import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { List, ListItem } from 'react-native-elements';
import { connect } from 'react-redux';

import { screenStyles } from '../../styles';
import { selectDecision } from '../../actions/deciderActions';
import { Header } from '../common';

const DecisionListComponent = ({ decisions, selectDecision, navigation }) => (
  <View style={screenStyles.view}>
    <Header>Decision List</Header>
    <List style={{ flex: 1 }}>
    {
      decisions.map((item, index) => (
        <ListItem key={index} title={item.name} onPress={() => {
          selectDecision(item);
          navigation.navigate('decisionDetail');
        }} />
      ))
    }
    </List>
  </View>
);

DecisionListComponent.propTypes = {
  decisions: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      options: PropTypes.arrayOf(PropTypes.string)
    })
  ),
  selectDecision: PropTypes.func.isRequired,
  navigation: PropTypes.object.isRequired
};

const mapStateToProps = ({ decisions }) => ({ decisions });

const mapDispatchToProps = dispatch => ({
  selectDecision: decision => dispatch(selectDecision(decision))
});

const DecisionList = connect(mapStateToProps, mapDispatchToProps)(DecisionListComponent);

export { DecisionList, DecisionListComponent };
