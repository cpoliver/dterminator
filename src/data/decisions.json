[{
  "id": "000",
  "name": "What to have for dinner",
  "options": [
    "kebab",
    "soup",
    "salad",
    "pizza",
    "curry"
  ]
}, {
  "id": "001",
  "name": "Which chore to do next",
  "options": [
    "vacuum",
    "wash dishes",
    "laundry",
    "clean bathroom"
  ]
}, {
  "id": "002",
  "name": "What type of drink to order next",
  "options": [
    "beer",
    "cider",
    "G&T",
    "single malt",
    "white russian",
    "espresso martini"
  ]
}, {
  "id": "003",
  "name": "What music to listen to",
  "options": [
    "drum 'n' bass",
    "hip hop",
    "jazz",
    "rock",
    "metal",
    "big beat",
    "house",
    "ambient"
  ]
}]
