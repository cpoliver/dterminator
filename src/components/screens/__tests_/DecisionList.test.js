import 'react-native';
import React from 'react';
import { ListItem } from 'react-native-elements';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import decisions from '../../../data/decisions.json';
import { DecisionListComponent as DecisionList } from '../DecisionList';

const selectDecisionStub = jest.fn();
const navigationStub = { navigate: jest.fn() };

const render = () => shallow(
  <DecisionList
    decisions={decisions}
    selectDecision={selectDecisionStub}
    navigation={navigationStub} />
);

describe('the DecisionList component', () => {
  it('should match the snapshot', () => {
    const component = render();

    expect(toJson(component)).toMatchSnapshot();
  });

  describe('when pressing a list item', () => {
    it('should call the relevant functions', () => {
      const listItem = render().find(ListItem).first();

      listItem.simulate('press');

      expect(selectDecisionStub).toHaveBeenCalledTimes(1);
      expect(navigationStub.navigate).toHaveBeenCalledTimes(1);
    });
  });
});
