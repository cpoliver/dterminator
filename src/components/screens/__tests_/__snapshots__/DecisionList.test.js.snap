// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the DecisionList component should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#f6f7fb",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <Header>
    DecisionList
  </Header>
  <List
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <ListItem
      chevronColor="#bdc6cf"
      hideChevron={false}
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      roundAvatar={false}
      switchButton={false}
      textInputEditable={true}
      title="What to have for dinner"
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      hideChevron={false}
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      roundAvatar={false}
      switchButton={false}
      textInputEditable={true}
      title="Which chore to do next"
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      hideChevron={false}
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      roundAvatar={false}
      switchButton={false}
      textInputEditable={true}
      title="What type of drink to order next"
      underlayColor="white"
    />
    <ListItem
      chevronColor="#bdc6cf"
      hideChevron={false}
      onPress={[Function]}
      rightIcon={
        Object {
          "name": "chevron-right",
        }
      }
      roundAvatar={false}
      switchButton={false}
      textInputEditable={true}
      title="What music to listen to"
      underlayColor="white"
    />
  </List>
</View>
`;
